### ISSUE TITLE
(Vote) Fixing ambiguity in current V3

### ISSUE THREAD

**Issue by koresar** [stampit-org/stampit#232]
In current stampit V3 RC if you import like that
```js
import {compose} from 'stampit';
```
the `compose` would be pure standard `compose` function from the specification (no stampit additional features). 

But, if you are using CommonJS
```js
const {compose} = require('stampit');
```
then the `compose` will be the infected one (all stampit features included).

_We need to consolidate the exports._

=====

### Solution 0
Leave as is.

Pros: 0 effort implementation.
Cons: Inconsistent exports in ES5 and ES6.

### Solution 1
Leave the CJS implementation to export only the stampit without the "pure `compose`", `isStamp`, and `isComposable`. Add the ability to import those three as `require('stampit/compose')`, `require('stampit/isStamp')`, `require('stampit/isComposable')`.

Pros: The least confusing way for the enduser.
Cons: Little harder to maintain the module.

### Solution 2
Do not export "pure `compose`" from CJS.

Pros: Very simple implementation.
Cons: only stampit-infected stamps can be created.

### Solution 3
Export "pure `compose`" from CJS as `stampit.compose`.

Pros: ?
Cons: Really inconsistent. `stampit.compose()` would produce pure stamps. But, say, `stampit.init()` and the rest would produce the infected stamps.

### Solution 4
Export "pure `compose`" from CJS as `stampit.compose`, along with the `.isStamp`, and `.isComposable`. But!

Remove all the shortcut functions like `const {init} = require('stampit')`. The only way to get those would be `const {init} = require('stampit')()`.

Pros: Simplifies everything.
Cons: That way of importing a shortcut function can be weird for some people.

## Solution 5
Export "pure `compose`" from CJS as `stampit.pureCompose`.

Pros: 1 line implementation
Cons: New function name, new API, new docs.

### Solution 6 - Your proposal?

=====

### Vote below

**Comment by FredyC** [comment]
I would surely go with number 2 and perhaps include number 3 with modification that could be exported as `stampit.pureCompose` just to prevent confusion. And I think that number 4 is rather ugly breaking change, I am certainly using those separate exports all over the place and most probably I am not alone.

**Comment by koresar** [comment]
I'll add your proposal to the list above.

Btw, I personally like Solution 1.

**Comment by FredyC** [comment]
I guess that number 1 has it's charms too, I just hate to keep source files floating in root directory. Although this could be solved by changing build process slightly and instead of publishing from root, it's possible to publish from subdirectory. Those files could then be just copied there, otherwise they could be kept in src.

**Comment by koresar** [comment]
Yeah. That's what I meant saying "Little harder to maintain the module." :)

The issue with "pureCompose" is that only you and me (and maybe few more people) know what that is. Most people would be confused and would need to learn "infected" and "pure" concepts we have invented. :)

Thus, I would better take some time implementing Solution 1. Hope that's fine.

**Comment by koresar** [comment]
Implemented as #233

**Comment by koresar** [comment]
Solved.
